### PR TITLE
Allow views in api keys

### PR DESCRIPTION
**Pull request by oleurud**
Fix: https://github.com/CartoDB/support/issues/1776

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Development
 - None yet
 
 ### Bug fixes / enhancements
-- None yet
+- Allowing views in API Keys
```
Comment (javitonino) [inline NEWS.md]: As mentioned in other PR, please add a reference to issue/PR.

Comment (oleurud) [reply NEWS.md]: Ok. I will add a test with a materialized view

File: spec/models/carto/api_key_spec.rb [hunk 2]
```diff
@@ -159,6 +159,43 @@ def with_connection_from_api_key(api_key)
...
+      )
+
+      validate_view_api_key(
+        view_name,
+        "CREATE MATERIALIZED VIEW #{view_name} AS SELECT * FROM #{@table1.name}",
+        "DROP MATERIALIZED VIEW #{view_name}"
+      )
+    end
+
+    def validate_view_api_key(view_name, createQuery, dropQuery)
```
[the remark this answers is not in the document]
Comment (javitonino) [reply spec/models/carto/api_key_spec.rb]: node style is leaking :see_no_evil:

[the remark this answers is not in the document]
Comment (oleurud) [reply spec/models/carto/api_key_spec.rb]: goat always goes to the woods  (spanish saying)

**Comment by oleurud** [comment]
@javitonino I really needed to add a new check looking into `pg_matviews` to support materialized views. Now it is ready for a new CR

**Pull request merged**